tests/test_curate_agentic.py

```
@@ -0,0 +1,484 @@
+        self.assertIn(REASON_SKIPPED_KIND, decision["reason_codes"])
+
+    def test_transform_is_output_idempotent(self):
+        source = episode_fixture(steps=[_step(1, thought="scratch")])
+        once, _ = curate_record(source)
+        twice, second = curate_record(once)
+        self.assertEqual(once, twice)
+        self.assertEqual(second["action"], ACTION_RETAINED)
+        self.assertEqual(second["thought_fields_removed"], 0)
+
+    def test_curate_source_scans_tree_and_handles_empty(self):
+        empty = curate_source(Path("/tmp/curate-agentic-missing-source-does-not-exist"))
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/curate-agentic-missing-source-does-not-exist"

(S108)